src/java.base/share/classes/java/util/HexDigits.java

```diff
@@ -39,38 +39,27 @@ final class HexDigits implements Digits {
      * Each element of the array represents the ascii encoded
      * hex relative to its index, for example:<p>
      * <pre>
-     *       0 -> '00' -> ('0' << 8) | '0' -> 12336
-     *       1 -> '01' -> ('0' << 8) | '1' -> 12337
-     *       2 -> '02' -> ('0' << 8) | '2' -> 12338
+     *       0 -> '00' -> '0' | ('0' << 8) -> 0x3030
```

Comment:
AFAICT this is now a copy of `StringLatin1::PACKED_DIGITS`, which might be fine. The comment to `StringLatin1::PACKED_DIGITS` seem wrong, though? I think that needs a re-examination (in this or in a separate PR)

Reply:
Perhaps you could extract the `digitPair` method you're adding in #15555 to a separate PR and use that to remove `HexDigits::DIGITS`?

Reply:
Please help me create an issue, and I will submit a new PR to modify it.

Reply:
Filed https://bugs.openjdk.org/browse/JDK-8315968 - we should also get @JimLaskey to weigh in here (who's recently authored the `java.util.Digits` code). Tl;dr: I think it'd be good to consolidate and share the code to produce these packed digit pairs.

Also note that there might be some issues on big-endian systems with recent changes, so let's figure that out first: https://bugs.openjdk.org/browse/JDK-8310929

Reply:
https://github.com/wenshao/jdk/tree/fix_14699_big_endian

I have a [branch](https://github.com/wenshao/jdk/tree/fix_14699_big_endian) ready to fix the big endian issue and need someone to create an issue

Reply:
https://github.com/openjdk/jdk/pull/15652

I've submitted a PR to fix the big endian problem